Plutarch/Api/V1/AssocMap.hs

```diff
@@ -378,32 +376,30 @@ punionWithData = phoistAcyclic $
     pcon $ PMap $ (pmatch (mapUnion # combine) \(MapUnionCarrier {merge}) -> merge) # pto x # pto y
 
 -- | Difference of two maps. Return elements of the first map not existing in the second map.
-pdifference ::
-  (PIsData k, PIsData a, PIsData b) =>
-  Term s (PMap g k a :--> PMap _ k b :--> PMap g k a)
+pdifference :: PIsData k => Term s (PMap g k a :--> PMap any k b :--> PMap g k a)
 pdifference = phoistAcyclic $
   plam $ \left right ->
     pcon . PMap $
       precList
         ( \self x xs ->
             plet (self # xs) $ \xs' ->
               pfoldAt
-                # (pfromData $ pfstBuiltin # x)
```

Comment:
all the minor syntax changes was because I got fed up of hlint screaming at me while working on these files

could have a `hlint.yaml` with your preferences in the repo